pkg/azuredx/client/client.go

```diff
@@ -18,23 +19,29 @@ import (
 
 type AdxClient interface {
 	TestRequest(ctx context.Context, datasourceSettings *models.DatasourceSettings, properties *models.Properties, additionalHeaders map[string]string) error
-	KustoRequest(ctx context.Context, url string, payload models.RequestPayload, additionalHeaders map[string]string) (*models.TableResponse, error)
+	KustoRequest(ctx context.Context, cluster string, url string, payload models.RequestPayload, additionalHeaders map[string]string) (*models.TableResponse, error)
+	ARGClusterRequest(ctx context.Context, payload models.ARGRequestPayload, additionalHeaders map[string]string) ([]models.ClusterOption, error)
 }
 
 var _ AdxClient = new(Client) // validates interface conformance
 
 // Client is an http.Client used for API requests.
 type Client struct {
-	httpClient *http.Client
+	httpClientAzureCloud *http.Client
+	httpClientManagement *http.Client
```

Comment:
It would be nice to not have 2 clients, but I could not think of a way around this. Any ideas?